src/components/Utils/helpers.js

```diff
@@ -37,3 +37,52 @@ export const latestInterval = (
 export const isSubstring = (fullText = '', textToFind = '') => {
   return fullText.toLowerCase().includes(textToFind.toLowerCase());
 };
+
+export function differenceInDays(dateOne, dateTwo) {
+  const difference = dateOne.getTime() - dateTwo.getTime();
+  const days = Math.ceil(difference / (1000 * 3600 * 24));
+  return days;
+}
+
+export const getDaysNextPurchase = (product) => {
+  const havePurchases = product.numberOfPurchases > 0;
+
+  const estimatedDays = havePurchases
+    ? product.estimatedDaysNextPurchase
+    : product.nextPurchase;
+  const lastPurchase = havePurchases
+    ? product.lastPurchasedDate.toDate()
+    : product.creationDate.toDate();
+
+  const dateNextPurchase = new Date(lastPurchase);
+  dateNextPurchase.setDate(lastPurchase.getDate() + estimatedDays);
+
+  const today = new Date();
+  const daysNextPurchase = differenceInDays(dateNextPurchase, today);
+
+  return daysNextPurchase;
+};
+
+export const sortProducts = (p1, p2) => {
+  let order = p1.daysNextPurchase - p2.daysNextPurchase;
+  if (order === 0) order = p1.item > p2.item ? 1 : -1;
+
+  return order;
+};
+
+export function getProductStatus(product) {
+  let status = '';
+  const { daysNextPurchase, estimatedDaysNextPurchase, nextPurchase } = product;
+  const havePurchases = product.numberOfPurchases > 0;
+  const estimatedDays = havePurchases
+    ? estimatedDaysNextPurchase
+    : nextPurchase;
+
+  if (daysNextPurchase > 2 * estimatedDays) status = 'inactive';
```

Comment:
Status: Inactive

Well, when I wanted to check an 'inactive' status, I couldn't it.

This is my example:

Property| Value
--- | --- 
`token`| `alejandro-test`
`item` | `papaya-soon`
`estimatedDaysNextPurchase` | `6`
`lastPurchasedDate` 1st. purchase | `12/02/2021` (today)
`lastPurchasedDate` 1st. purchase (modified for testing) | `31/01/2021` (return 2*6 days in the past time)

but, status is still soon (or **_green_**), maybe I'm wrong but I believed it will be '`inactive`'

Please, could you give me some clarification? I'll appreciate it 🙏 .

Reply:
I will check, thank you very much

Reply:
Hi @alxmcr thanks a lot, there was a bug with the inactive status. It's fixed, can you try again the test you made. Something to keep in mind, there is a new field on the database named `creationDate`, if an item that have not been purchased it's too old, it will also got inactive, try this: 

- Add a product "soon"
- Then go to te database and change the creation Date to 01/02/2021. 
- Check the item list, you'll see that this product is inactivated.

Thanks a lot for finding this bug.

Reply:
You're welcome, Mariangel 😄 !

Thanks for your suggestions! I followed your instructions and you're right, you fixed the bug. Congratulations! 👍

Only, I'm wondering if it's necessary, the inactive products should be alphabetically sorted.